packages/react-router/src/router.ts

```diff
@@ -2675,7 +2674,7 @@ export class Router<
     )
 
     this.streamedKeys.add(key)
-    const children = `__TSR__.streamedValues['${key}'] = { value: ${this.serializer?.(this.options.transformer.stringify(value))}}`
+    const children = `__TSR__.streamedValues['${key}'] = ${jsesc({ value: this.serializer?.(this.options.transformer.stringify(value)) }, { isScriptContext: true, wrap: true, json: true })}`
```

Comment:
What's the purpose of this change?

Reply:
Depending on the transformer/serializer used, the streamed value may still contain illegal characters that need to be escaped.

Reply:
Are you running into this issue?

I'm asking since the `defaultTransformer` handles `undefined` and `Date` values already.

If you aren't running this issue, I'd rather not add this package into core, that'd have a downstream effect on SPA users as well.

Reply:
Yep, I tried to use @tanstack/react-router-with-query - I am using `devalue` as a transformer b/c there are some values i.e. `Decimal` that cannot be serialized with the default transformer. Using `devalue` as a transformer causes illegal values to be injected into the <script/> tag which causes hydration to fail.